### DISCUSSION TITLE
What does this Deno Worker error means?

### DISCUSSION THREAD

**Discussion by sinyo-matu** [denoland/deno#15205] [Q&A] [answered]
Hi everyone.
I am working on an aleph.js app
And tried the Deno Worker API first time. 
Any one know about this error?

I'm so confusing about this error happens immediately after app started, 
even before the postMessage() method called in my code.

And I just CAN NOT catch it anywhere of my code.

```shell
error: Uncaught (in worker "") TypeError: Failed to execute 'postMessage' on 'DedicatedWorkerGlobalScope': Argument 2 can not be converted to a dictionary
    at makeException (deno:ext/webidl/00_webidl.js:80:12)
    at Array.StructuredSerializeOptions (deno:ext/webidl/00_webidl.js:682:17)
    at postMessage (deno:runtime/js/99_main.js:118:35)
    at https://esm.sh/v86/docx@7.4.0/deno/docx.js:3:89683
    at https://esm.sh/v86/docx@7.4.0/deno/docx.js:3:89720
    at Object.<anonymous> (https://esm.sh/v86/docx@7.4.0/deno/docx.js:3:90875)
    at Object.ct.exports.54 (https://esm.sh/v86/docx@7.4.0/deno/docx.js:3:90918)
    at o (https://esm.sh/v86/docx@7.4.0/deno/docx.js:2:34875)
    at https://esm.sh/v86/docx@7.4.0/deno/docx.js:2:34909
    at Object.ct.exports.32../base64 (https://esm.sh/v86/docx@7.4.0/deno/docx.js:3:25275)
error: Uncaught (in promise) Error: Unhandled error in child worker.
    at Worker.#pollControl (deno:runtime/js/11_workers.js:152:21)
```

And In my code,
I call postMessage like
```typescript
// SendFaxChannelMessage is a Dictionary object
export function sendStartTask(worker: Worker, message: SendFaxChannelMessage) {
  worker.postMessage(message);
}
```

**Comment by nayeemrmn** [comment, marked as answer]
It looks like docx is calling `postMessage("", "*")` which does also fail on the web.

**Reply by sinyo-matu** [reply]
Thanks for you response!
but the second arg is Transferable[] type.

**Reply by nayeemrmn** [reply]
`Transferable[]` needs to be an array of `ArrayBuffer | MessagePort`

**Reply by andreubotella** [reply]
(In browsers there are other transferable types, such as `File`, but Deno doesn't support them yet: #12067)

**Reply by sinyo-matu** [reply]
Update my question. I'm even not offering second arg

**Reply by nayeemrmn** [reply]
The library you're using is at `https://esm.sh/v86/docx@7.4.0/deno/docx.js:3:89683`

**Reply by sinyo-matu** [reply]
> The library you're using is at `https://esm.sh/v86/docx@7.4.0/deno/docx.js:3:89683`

Oh.. Thank you! I got it.
